### PR TITLE
#51 | Added Bitbucket Oauth2 support

### PR DESCRIPTION
**Pull request by afdecastro879**
This PR adds Bitbucket oauth2 as a provider

### CODE REVIEW

File: oauth2/bitbucket.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+// Using the avatar url to be able to fetch 128px image. By default BitbucketAPI return 32px image.
+var bitbucketAvatarURL = "https://bitbucket.org/account/%v/avatar/128/"
+
+func init() {
+	RegisterProvider(providerBitbucket)
+}
+
+// BitbucketUser is used for parsing the github response
+type BitbucketUser struct {
```
Comment (smancke) [inline oauth2/bitbucket.go]: Could also be private (lower case)

**Comment by smancke** [comment]
Hey @afdecastro879, thanks for the clean and tested PR!
I like the testify/suite, which was new to me. 
The coverage decreases a bit, but I think, it's not worth to test all the 'err != nil' statements.

**Comment by afdecastro879** [comment]
Hey @smancke Thank you for your feedback. I already sent a new commit fixing the private struct.

**Comment by smancke** [comment]
ok, cool!

**Pull request merged**